yktrace/src/hwt/mapper.rs

```diff
@@ -41,21 +41,29 @@ pub struct BlockMap {
 impl BlockMap {
     /// Parse the LLVM blockmap section of the current executable and return a struct holding the
     /// mappings.
-    ///
-    /// PERF: See if we can get the block map section marked as loadable so that the linker loads
-    /// it automatically at process creation time.
     pub fn new() -> Self {
-        let mut elems = Vec::new();
-        let pathb = env::current_exe().unwrap();
-        let file = fs::File::open(&pathb.as_path()).unwrap();
-        let mmap = unsafe { memmap2::Mmap::map(&file).unwrap() };
-        let object = object::File::parse(&*mmap).unwrap();
-        let sec = object.section_by_name(BLOCK_MAP_SEC).unwrap();
-        let sec_size = sec.size();
-        let mut crsr = Cursor::new(sec.data().unwrap());
+        let (bc, bc_len) = llvmbc_section();
 
-        // Keep reading records until we fall outside of the section's bounds.
-        while crsr.position() < sec_size {
+        let mut start_addr = MaybeUninit::uninit();
+        let mut end_addr = MaybeUninit::uninit();
+        let found = unsafe {
+            __ykllvmwrap_find_bbmaps(bc, bc_len, start_addr.as_mut_ptr(), end_addr.as_mut_ptr())
+        };
+        if !found {
+            panic!("couldn't find .llvmbbaddrmap data!");
+        }
+        let (start_addr, end_addr) = unsafe { (start_addr.assume_init(), end_addr.assume_init()) };
+        let bbaddrmap_data =
+            unsafe { slice::from_raw_parts(start_addr as *const u8, end_addr - start_addr) };
+
+        debug_assert!(start_addr != usize::MAX);
+        debug_assert!(end_addr != usize::MIN);
+        debug_assert!(end_addr >= start_addr);
```

Comment:
Should this be `>`?

Reply:
I think you might be right. A binary must have at least one block, as evidenced by:
```
$ clang /dev/null
ld: error: undefined symbol: main
>>> referenced by /usr/lib/crt0.o:(.text+0x13E)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
I think having the stronger assert (i.e. `>`) as a starting point can't hurt us.

Reply:
Fixed in [2167cc6](https://github.com/ykjit/yk/pull/571/commits/2167cc63d49395be51a1f10ddd11049af803e056).